[restclient-cpp] Fix portfile and update version.

et-nik opened the pull request:
This PR to update restclient library and fix portfile. Previously, the portfile had a command to delete the all compiled files (`restclient-cpp.lib`, `restclient-cppd.lib`), it is mistake.

I update library version from 0.5.1 tag to last commit. CMakeLists patches are no longer required in the new library version.
Library is updated and bugfixes, but the developer did not add new tags. Some people want tag updating (https://github.com/mrtazz/restclient-cpp/issues/145).

et-nik committed:
[restclient-cpp] Fix installation. Remove only debug directory

et-nik committed:
[restclient-cpp] Fix directory cleanup

et-nik committed:
[restclient-cpp] Update library

et-nik committed:
[restclient-cpp] Add msvc patch

et-nik committed:
[restclient-cpp] Remove lib directory on dynamic compile

et-nik committed:
[restclient-cpp] set VCPKG_POLICY_DLLS_WITHOUT_LIBS

et-nik committed:
[restclient-cpp] Remove lib directory on dynamic compile

et-nik committed:
[restclient-cpp] Set compile type static

et-nik committed:
Remove patch

NancyLi1013 commented on ports/restclient-cpp/portfile.cmake:
```diff
@@ -3,27 +3,28 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
`include(vcpkg_common_functions)` is no longer needed.
Could you remove this line?

et-nik replied:
Removed

NancyLi1013 commented on ports/restclient-cpp/portfile.cmake:
```diff
@@ -3,27 +3,28 @@ include(vcpkg_common_functions)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 # Handle copyright
 file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/restclient-cpp)
```
Could you update 
```
file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/restclient-cpp)
file(RENAME ${CURRENT_PACKAGES_DIR}/share/restclient-cpp/LICENSE ${CURRENT_PACKAGES_DIR}/share/restclient-cpp/copyright)
```
as
`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`?

NancyLi1013 commented on ports/restclient-cpp/portfile.cmake:
```diff
@@ -3,27 +3,28 @@ include(vcpkg_common_functions)
-        0001_fix_cmake_linking.patch
 )
 
+set(COMPILE_TYPE STATIC)
```
Why do you set `COMPILE_TYPE `as `STATIC ` here ?

et-nik replied:
If `COMPILE_TYPE` is `STATIC` set, it will be compiled lib file.
If `COMPILE_TYPE` not set, it will be compiled dll file.

It is not CRT parameter. And does not affect CRT compile parameters.

See https://github.com/mrtazz/restclient-cpp/blob/master/CMakeLists.txt#L46

I moved from here to cmake options `-DCOMPILE_TYPE=STATIC`

NancyLi1013 commented:
Hi @et-nik Thanks for this PR.
Could you also bump the version in `CONTROL `file?

et-nik committed:
[restclient-cpp] Changes and fixes
Update version
Update handling copiright
Remove no longer needed include
Move COMPILE_TYPE parameter

et-nik commented:
Version changed in `CONTROL` file.

`vcpkg-osx-PR-test` still failed. I think this fail is not related port and package.

NancyLi1013 commented:
/azp run

et-nik commented:
Hm. Now _vcpkg-windows-PR-test_ failed

et-nik commented:
/azp run

et-nik commented:
@NancyLi1013 could you run tests again?

NancyLi1013 commented:
@et-nik thanks for your remind.
The PR failed on windows platform due to` ppconsul`.
PR #9752 will fix the problem.
So let's wait for PR #9752 merged, then I will re run this PR again.

vicroms commented on ports/restclient-cpp/portfile.cmake:
```diff
@@ -16,18 +12,18 @@ vcpkg_configure_cmake(
     OPTIONS
         -DCMAKE_DISABLE_FIND_PACKAGE_GTest=TRUE
         -DCMAKE_DISABLE_FIND_PACKAGE_jsoncpp=TRUE
+        -DCOMPILE_TYPE=STATIC
```
Why is this necessary? Does it affect dynamic library builds?

et-nik replied:
See https://github.com/microsoft/vcpkg/pull/9717#discussion_r367821525

This is not the same `static` triplets vcpkg thing.

Default `COMPILE_TYPE` is SHARED. It is mean than only .dll/.so library will be compile.
I agree that this is a little confusing. I can add CMakeLists.txt patch instead of this option.

et-nik replied:
@vicroms Restclient `CmakeList.txt` is made in such a way that it is compile to either lib or dll.
We can't way to build both (lib and dll).
Without `-DCOMPILE_TYPE=STATIC`, only dynamic library will be compiled.
I think it's better to build static libraries, or both. Patch `CmakeList.txt` is needed if we want build static and dynamic libraries.

strega-nil replied:
@ras0219-msft What's the opinion of vcpkg here? ^

dan-shaw replied:
We should use `VCPKG_LIBRARY_LINKAGE`
See https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries

et-nik replied:
Okay. Thanks.
I'll fix. Also I'll bump restclient version. 0.5.2 release is available

maksqwe commented on ports/restclient-cpp/CONTROL:
```diff
@@ -1,4 +1,4 @@
-Version: 0.5.1-2
+Version: 0.5.1-3
 Build-Depends: curl
 Description: Binn is a binary data serialization format designed to be compact, fast and easy to use.
```
I think it is incorrect description, could you fix it?

et-nik replied:
Fixed. Also I added Homepage info

et-nik committed:
[restclient-cpp] Update description. Add Homepage

et-nik committed:
Merge branch 'master' into hotfix/restclient-cpp-fix

NancyLi1013 commented:
@et-nik 
Could you please address the review suggestion?

et-nik commented:
Hello. I'll update this PR today.

et-nik committed:
[restclient-cpp] Bump to 0.5.2. Remove COMPILE_TYPE option

NancyLi1013 commented:
It seems that the fix is not correct. Please refer to this [doc](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries) to try it again.

et-nik committed:
[restclient-cpp] Set ONLY_STATIC_LIBRARY for Windows

et-nik commented:
Done 

* Bumped version to last tag 0.5.2
* Removed old patches
* Fixed descriptions

NancyLi1013 commented on ports/restclient-cpp/CONTROL:
```diff
@@ -1,4 +1,5 @@
 Source: restclient-cpp
-Version: 0.5.1-3
+Version: 0.5.2-1
```
The version should be `0.5.2` instead of` 0.5.2-1`.
Could you please update it again?

et-nik committed:
[restclient-cpp] Change version num 0.5.2-1 -> 0.5.2

et-nik commented:
@NancyLi1013 Done

Hm. I see that all checks are succeeded https://dev.azure.com/vcpkg/public/_build/results?buildId=36451&view=results

But here check statuses are still "Waiting for..."

strega-nil commented:
Cool, looks good to me! Thanks @et-nik :)
